tests/components/yolink/test_config_flow.py

```diff
@@ -0,0 +1,202 @@
+
+
+async def test_abort_if_authorization_timeout(hass, current_request_with_host):
+    """Check yolink authorization timeout."""
+    assert await setup.async_setup_component(
+        hass,
+        DOMAIN,
+        {
+            DOMAIN: {
+                CONF_CLIENT_ID: CLIENT_ID,
+                CONF_CLIENT_SECRET: CLIENT_SECRET,
+            }
+        },
+    )
+
+    with patch(
+        "homeassistant.components.yolink.config_entry_oauth2_flow."
+        "LocalOAuth2Implementation.async_generate_authorize_url",
+        side_effect=asyncio.TimeoutError,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "authorize_url_timeout"
+
+
+async def test_reauthentication(
+    hass, hass_client_no_auth, aioclient_mock, current_request_with_host
+):
+    """Test yolink reauthentication."""
+    await setup.async_setup_component(
+        hass,
+        DOMAIN,
+        {
+            DOMAIN: {CONF_CLIENT_ID: CLIENT_ID, CONF_CLIENT_SECRET: CLIENT_SECRET},
+            "http": {"base_url": "https://example.com"},
+        },
+    )
+
+    old_entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=123,
+        version=1,
+        data={"id": "frenck", "auth_implementation": DOMAIN},
+    )
+    old_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": config_entries.SOURCE_REAUTH,
+            "unique_id": old_entry.unique_id,
+            "entry_id": old_entry.entry_id,
+        },
+        data=old_entry.data,
+    )
+
+    flows = hass.config_entries.flow.async_progress()
+    assert len(flows) == 1
+
+    result = await hass.config_entries.flow.async_configure(flows[0]["flow_id"], {})
+
+    # pylint: disable=protected-access
+    state = config_entry_oauth2_flow._encode_jwt(
+        hass,
+        {
+            "flow_id": result["flow_id"],
+            "redirect_uri": "https://example.com/auth/external/callback",
+        },
+    )
+    client = await hass_client_no_auth()
+    await client.get(f"/auth/external/callback?code=abcd&state={state}")
+
+    aioclient_mock.post(
+        OAUTH2_TOKEN,
+        json={
+            "refresh_token": "mock-refresh-token",
+            "access_token": "mock-access-token",
+            "type": "Bearer",
+            "expires_in": 60,
+        },
+    )
+
+    with patch("homeassistant.components.yolink.api.ConfigEntryAuth"):
+        with patch(
+            "homeassistant.components.yolink.async_setup_entry", return_value=True
+        ) as mock_setup:
+            result = await hass.config_entries.flow.async_configure(result["flow_id"])
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "reauth_successful"
+    assert len(mock_setup.mock_calls) == 1
```

Comment:
Also assert that the existing config entry data is updated after the re-authentication.

Reply:
actually, no config entry data will update by re-authentication flow, I will remove that from test flow

Reply:
The tokens stored in the config entry data will be updated. That's the point of the re-authentication.

Reply:
Yeah，i known that,but i found some of other integration did't check the token update,i will add assert for token update check